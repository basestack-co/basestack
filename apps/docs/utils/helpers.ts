export const repoUrl = "https://github.com/Storytics/monolock";
export const domainUrl = "https://basestack.co";
