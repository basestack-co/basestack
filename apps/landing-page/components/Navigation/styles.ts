import styled, { css, createGlobalStyle } from "styled-components";
import {
  Button,
  ButtonVariant,
  ButtonSize,
  IconButton,
} from "@basestack/design-system";
import { rem } from "polished";

export const GlobalStyle = createGlobalStyle<{ isMenuOpen: boolean }>`
  ${({ isMenuOpen }) =>
    isMenuOpen &&
    css`
      body {
        overflow: hidden;
      }
    `}
`;

const flexColumn = css`
  display: flex;
  flex-direction: column;
`;

const backgroundBlur = (isDarkMode: boolean) => css`
  background: ${isDarkMode
    ? "rgba(31, 31, 31, 0.6)"
    : "rgba(238, 238, 238, 0.7)"};
  -webkit-backdrop-filter: saturate(180%) blur(20px);
  backdrop-filter: saturate(180%) blur(20px);
`;

export const Container = styled.nav<{
  isDarkMode: boolean;
  hasBackdropFilter: boolean;
}>`
  position: sticky;
  top: 0;
  z-index: 1000;
  display: flex;
  flex-shrink: 0;
  align-items: center;
  height: ${rem("80px")};
  padding: 0 ${({ theme }) => theme.spacing.s5};
  transition: all 0.2s ease-in;

  ${({ hasBackdropFilter, isDarkMode, theme }) =>
    hasBackdropFilter
      ? backgroundBlur(isDarkMode)
      : css`
          background-color: ${isDarkMode
            ? theme.colors.gray800
            : theme.colors.gray50};
        `}
`;

export const ContentContainer = styled.div`
  display: flex;
  align-items: center;
  margin: 0 auto;
  width: 100%;
  max-width: 1440px;
`;

export const LeftColumn = styled.div`
  display: flex;
  align-items: center;
`;

export const List = styled.ul`
  display: flex;
  margin-left: ${({ theme }) => theme.spacing.s3};
`;

export const ListItem = styled.li`
  margin-right: ${({ theme }) => theme.spacing.s2};
`;

export const RightColumn = styled.div`
  display: flex;
  margin-left: auto;
`;

export const Logo = styled.div`
  display: inline-flex;
  justify-content: center;
  position: relative;
  height: ${rem("40px")};
  width: ${rem("40px")};
  border-radius: 50%;
  background-color: ${({ theme }) => theme.colors.black};

  &::before {
    content: "";
    position: absolute;
    bottom: ${rem("5px")};
    border-radius: 50%;
    height: ${rem("20px")};
    width: ${rem("20px")};
    background-color: ${({ theme }) => theme.colors.white};
  }
`;

export const PopupContainer = styled.div<{ isMenuOpen: boolean }>`
  position: fixed;
  z-index: 999;
  top: 79px;
  left: 0;
  right: 0;
  bottom: 0;
  overflow: hidden;
  ${flexColumn};
  align-items: center;
  justify-content: center;

  ${({ isMenuOpen }) =>
    !isMenuOpen &&
    css`
      pointer-events: none;
    `};
`;

export const PopupWrapper = styled.ul<{ isDarkMode: boolean }>`
  ${flexColumn};
  ${({ isDarkMode }) => backgroundBlur(isDarkMode)};
  padding: ${({ theme }) => theme.spacing.s3};
`;

export const PopupItem = styled.li`
  ${flexColumn};

  &:not(:last-child) {
    margin-bottom: ${({ theme }) => theme.spacing.s3};
  }
`;

export const BurgerMenu = styled(IconButton)<{ isDarkMode: boolean }>`
  ${({ isDarkMode }) =>
    isDarkMode &&
    css`
      background-color: transparent;
      color: ${({ theme }) => theme.colors.gray300};
      transition: background-color 0.1s ease-in-out;

      &:hover {
        background-color: ${({ theme }) => theme.colors.gray700};
      }
    `}
`;
