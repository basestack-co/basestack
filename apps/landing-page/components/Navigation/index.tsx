import React, { useState } from "react";
// Router
import { useRouter } from "next/router";
// Utils
import { config as defaults, events } from "@basestack/utils";
// Theme
import { useTheme } from "styled-components";
import { useMediaQuery } from "@basestack/hooks";
import {
  useTransition,
  useSpring,
  useChain,
  config,
  animated,
  useSpringRef,
} from "react-spring";
// Components
import {
  Button,
  ButtonVariant,
  ButtonSize,
  Logo,
  IconButton,
} from "@basestack/design-system";
import {
  Container,
  ContentContainer,
  LeftColumn,
  List,
  ListItem,
  PopupContainer,
  PopupWrapper,
  PopupItem,
  RightColumn,
  GlobalStyle,
  BurgerMenu,
} from "./styles";

const links = [
  {
    text: "Platform",
    href: "#platform",
  },
  {
    text: "Discover Features",
    href: "#features",
  },
  {
    text: "Integration",
    href: "#code",
  },
  {
    text: "Why Feature Flags?",
    href: "#why",
  },
  {
    text: "FAQs",
    href: "#questions",
  },
];

interface NavigationProps {
  isDarkMode: boolean;
  hasBackdropFilter?: boolean;
}

const AnimatedPopup = animated(PopupWrapper);
const AnimatedPopupItem = animated(PopupItem);

const Navigation = ({
  isDarkMode,
  hasBackdropFilter = true,
}: NavigationProps) => {
  const theme = useTheme();
  const [isMenuOpen, setIsMenuOpen] = useState(false);
  const router = useRouter();

  const springApi = useSpringRef();
  const { size, ...rest } = useSpring({
    ref: springApi,
    config: config.stiff,
    from: { size: "0", opacity: 0 },
    to: {
      size: isMenuOpen ? "100%" : "0",
      opacity: isMenuOpen ? 1 : 0,
    },
  });

  const transApi = useSpringRef();
  const transition = useTransition(isMenuOpen ? links : [], {
    ref: transApi,
    trail: 400 / links.length,
    from: { opacity: 0 },
    enter: { opacity: 1 },
    leave: { opacity: 0 },
  });

  useChain(isMenuOpen ? [springApi, transApi] : [transApi, springApi], [
    0,
    isMenuOpen ? 0.3 : 0.6,
  ]);

  return (
    <>
      <GlobalStyle isMenuOpen={isMenuOpen} />
      <Container isDarkMode={isDarkMode} hasBackdropFilter={hasBackdropFilter}>
        <ContentContainer>
          <LeftColumn>
            {/* Mobile Burger */}
            <BurgerMenu isDarkMode={isDarkMode}>
              <IconButton
                size="large"
                icon={isMenuOpen ? "menu_open" : "menu"}
                onClick={() => setIsMenuOpen((prevState) => !prevState)}
                mr={theme.spacing.s3}
              />
            </BurgerMenu>
            <Logo />
            {/* Desktop List */}
            <List>
              {links.map((link, index) => {
                return (
                  <ListItem key={index.toString()}>
                    <Button
                      variant={
                        isDarkMode
                          ? ButtonVariant.Secondary
                          : ButtonVariant.Tertiary
                      }
                      onClick={() => {
                        events.landing.navigation(link.text, link.href);
                        router.push(link.href);
                      }}
                      size={ButtonSize.Medium}
                      backgroundColor="transparent"
                    >
                      {link.text}
                    </Button>
                  </ListItem>
                );
              })}
            </List>
          </LeftColumn>
          <RightColumn>
            <Button
              variant={
                isDarkMode ? ButtonVariant.Secondary : ButtonVariant.Tertiary
              }
              mr={theme.spacing.s3}
              onClick={() => {
                if (typeof window !== "undefined") {
                  window.open(defaults.urls.repo, "_blank");
                }
              }}
              size={ButtonSize.Medium}
              backgroundColor="transparent"
            >
              Github
            </Button>
            <Button
              onClick={() => {
                if (typeof window !== "undefined") {
                  window.open(defaults.urls.docs.base, "_blank");
                }
              }}
              size={ButtonSize.Medium}
              variant={ButtonVariant.Secondary}
            >
              Documentation
            </Button>
          </RightColumn>
        </ContentContainer>
      </Container>
      {/* Mobile Menu */}
      <PopupContainer isMenuOpen={isMenuOpen}>
        <AnimatedPopup
          style={{ ...rest, width: size, height: size }}
          isDarkMode={isDarkMode}
        >
          {transition(
            (style, item) =>
              item && (
                <AnimatedPopupItem style={style}>
                  <Button
                    variant={
                      isDarkMode
                        ? ButtonVariant.Secondary
                        : ButtonVariant.Tertiary
                    }
                    onClick={() => {
                      setIsMenuOpen(false);
                      events.landing.navigation(item.text, item.href);
                      router.push(item.href);
                    }}
                    size={ButtonSize.Medium}
                    backgroundColor="transparent"
                    fullWidth
                  >
                    {item.text}
                  </Button>
                </AnimatedPopupItem>
              ),
          )}
        </AnimatedPopup>
      </PopupContainer>
    </>
  );
};

export default Navigation;
