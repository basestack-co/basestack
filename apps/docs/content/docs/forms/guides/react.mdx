---
title: NextJS & React
icon: Code
tags: Basestack Forms, NextJS, React Integration, Form submission, Fetch API, Axios, React, Next.js, API integration, JSON response, Web development, Developer tools, "2025"
---

import { Step, Steps } from 'fumadocs-ui/components/steps';

[React](https://react.dev/?ref=basestack.co) is a JavaScript library for building user interfaces, allowing developers to create interactive, dynamic web applications with reusable components. [Next.js](https://nextjs.org/?ref=basestack.co) is a React framework that enables server-side rendering, static site generation, and simplified routing for building fast, scalable web applications.

To get started, youâ€™ll need the following:

- A [Basestack Forms](https://forms.basestack.co) Account
- A React or Next.js project with a Form component

<Steps>

<Step>
### Use this on your React Project to send form data to your endpoint.

Send data using the [Fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API?ref=basestack.co), [Axios](https://axios-http.com/?ref=basestack.co) or Any [HTTP](https://developer.mozilla.org/en-US/docs/Web/HTTP?ref=basestack.co) Client and ensure the mode on the query string is set to `mode=rest` for retrieving the response in JSON format.
</Step>
<Step>
### Copy and paste this into your React component.

Feel free to use the code below to send a test submission.


```tsx copy title="contact.tsx"
import React, { useState } from "react";

const FormComponent = () => {
  // You can also use rect-hook-form or formik for form handling
  // check out the docs for more info https://react-hook-form.com/
  const [name, setName] = useState("");
  const [email, setEmail] = useState("");
  const [message, setMessage] = useState("");

  const onHandleSubmit = (e) => {
    e.preventDefault();
    e.stopPropagation();

    // You can also use axios or any other library to make the request
    // You can also use async await to make the request
    fetch("https://forms.basestack.co/api/v1/s/[KEY]?mode=rest", {
      method: "POST",
      headers: {
        Accept: "application/json",
        "Content-Type": "application/json",
      },
      body: JSON.stringify({ name, email, message }),
    })
      .then((res) => res.json())
      .then((data) => {
        if (data.code === 200) {
          console.log("Form submitted successfully");
        }
      })
      .catch((error) => {
        console.log("Error submitting form", error);
      });
  };

  return (
    <form onSubmit={(e) => onHandleSubmit(e)}>
      <div>
        <label htmlFor="name">Name:</label>
        {/*  <!-- name each of your inputs as you wish --> */}
        <input
          type="text"
          id="name"
          name="name"
          required
          autoComplete="name"
          aria-label="Name"
          placeholder="Enter your name"
          value={name}
          onChange={(e) => setName(e.target.value)}
        />
      </div>

      <div>
        <label htmlFor="email">Email:</label>
        <input
          type="email"
          id="email"
          name="email"
          required
          autoComplete="email"
          aria-label="Email"
          placeholder="Enter your email"
          value={email}
          onChange={(e) => setEmail(e.target.value)}
        />
      </div>

      <div>
        <label htmlFor="message">Message:</label>
        <textarea
          id="message"
          name="message"
          required
          aria-label="Message"
          placeholder="Enter your message"
          value={message}
          onChange={(e) => setMessage(e.target.value)}
        ></textarea>
      </div>

      {/*  <!-- your other form fields go here --> */}

      <button type="submit">Submit</button>
    </form>
  );
};

export default FormComponent;
```
</Step>
</Steps>
