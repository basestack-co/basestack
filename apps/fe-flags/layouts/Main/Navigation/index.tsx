import { useCallback, useMemo } from "react";
import { useTheme } from "styled-components";
// Components
import { Avatar, Button, ButtonVariant } from "@basestack/design-system";
import { Container, List, ListItem, LogoContainer } from "./styles";
import { ButtonLink, MoreMenu, ProjectsMenu } from "./components";
// Router
import { useRouter } from "next/router";
import Link from "next/link";
// Types
import { inferQueryOutput } from "libs/trpc";
// Store
import {
  seIsCreateProjectModalOpen,
  seIstCreateFlagModalOpen,
} from "contexts/modals/actions";
// Hooks
import useModals from "hooks/useModals";
// Auth
import { useSession } from "next-auth/react";

export interface LinkItem {
  text: string;
  to: string;
  isExternal?: boolean;
}

const leftItems = [
  {
    text: "Features",
    to: "/[projectSlug]/flags",
  },
  {
    text: "Settings",
    to: "/[projectSlug]/settings/general",
  },
];

const rightItems = [
  {
    text: "Documentation",
    to: "/documentation",
  },
  {
    text: "Resources",
    to: "/resources",
  },
  {
    text: "Github",
    to: "https://github.com/",
    isExternal: true,
  },
];

interface NavigationProps {
  isDesktop: boolean;
  data?: inferQueryOutput<"project.all">;
}

const Navigation = ({ isDesktop, data }: NavigationProps) => {
  const theme = useTheme();
  const { data: session } = useSession();
  const { dispatch } = useModals();
  const router = useRouter();

  const projectSlug = router.query.projectSlug as string;

  const onRenderItems = useCallback(
    (items: LinkItem[], type: string) =>
      items.map(({ text, to, isExternal }, i) => {
        return (
          <ListItem key={`${type}-list-item-${i}`}>
            <ButtonLink
              href={to.replace("[projectSlug]", projectSlug)}
              isExternal={isExternal}
              isActive={router.pathname === to}
            >
              {text}
            </ButtonLink>
          </ListItem>
        );
      }),
    [router.pathname, projectSlug]
  );

  const onSelectProject = useCallback(
    (projectSlug: string) => {
      router.push({
        pathname: "/[projectSlug]/flags",
        query: { projectSlug },
      });
    },
    [router]
  );

  const projects = useMemo(
    () =>
      data?.projects.map((item) => {
        return {
          id: item.id,
          slug: item.slug,
          onClick: () => onSelectProject(item.slug),
          text: item.name,
        };
      }),
    [data, onSelectProject]
  );

  return (
    <Container data-testid="navigation">
      <List data-testid="navigation-left-ul">
        <ListItem>
          <Link href="/">
            <LogoContainer>
              <Avatar round={false} alt="user image" userName="Logo" />
            </LogoContainer>
          </Link>
        </ListItem>
        <ProjectsMenu
          onClickCreateProject={() =>
            dispatch(seIsCreateProjectModalOpen(true))
          }
          projectSlug={projectSlug}
          projects={projects ?? []}
        />
        {isDesktop && !!projectSlug && (
          <>
            {onRenderItems(leftItems, "left")}
            <ListItem ml={theme.spacing.s5}>
              <Button
                onClick={() => dispatch(seIstCreateFlagModalOpen(true))}
                variant={ButtonVariant.Primary}
              >
                Create flag
              </Button>
            </ListItem>
          </>
        )}
      </List>
      <List ml="auto" data-testid="navigation-right-ul">
        {isDesktop && <>{onRenderItems(rightItems, "right")}</>}
        {!isDesktop && <MoreMenu />}
        <ListItem ml={theme.spacing.s3}>
          <Avatar
            src={session?.user.image ?? ""}
            alt="user image"
            userName={session?.user.name ?? session?.user.email}
          />
        </ListItem>
      </List>
    </Container>
  );
};

export default Navigation;
