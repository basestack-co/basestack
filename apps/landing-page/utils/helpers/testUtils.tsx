import theme from "@basestack/design-system/theme/lightTheme";
// Styles
import isPropValid from "@emotion/is-prop-valid";
import { RenderOptions, render } from "@testing-library/react";
import React from "react";
import { StyleSheetManager, ThemeProvider } from "styled-components";

interface WithChildrenProps {
  children: React.ReactNode;
}

interface AllProvidersProps extends WithChildrenProps {
  initialState?: {};
}

const AllProviders = ({ children, initialState = {} }: AllProvidersProps) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const renderWithAllProviders = (
  ui: React.ReactElement,
  initialState?: {},
  options?: Omit<RenderOptions, "wrapper">,
) =>
  render(ui, {
    // @ts-ignore
    wrapper: (props) => <AllProviders initialState={initialState} {...props} />,
    ...options,
  });

const WithThemeProvider = ({ children }: WithChildrenProps) => (
  <ThemeProvider theme={theme}>
    <StyleSheetManager shouldForwardProp={isPropValid}>
      {children}
    </StyleSheetManager>
  </ThemeProvider>
);

const renderWithTheme = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "wrapper">,
  // @ts-ignore
) => render(ui, { wrapper: WithThemeProvider, ...options });

export { renderWithAllProviders, renderWithTheme };
