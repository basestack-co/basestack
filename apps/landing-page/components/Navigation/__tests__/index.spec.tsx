import { cleanup } from "@testing-library/react";
import { renderWithTheme } from "utils/helpers/testUtils";
import Navigation from "..";

jest.mock("react-use", () => ({
  ...jest.requireActual("react-use"),
  useMedia: jest.fn(() => false),
}));

jest.mock("next/router", () => ({
  useRouter: jest.fn(),
}));

describe.skip("Navigation tests", () => {
  afterEach(cleanup);

  test("should renders Navigation correctly", () => {
    const { asFragment } = renderWithTheme(<Navigation isDarkMode={false} />);

    expect(asFragment()).toMatchSnapshot();
  });

  test("should render Navigation in dark mode", () => {
    const { asFragment } = renderWithTheme(<Navigation isDarkMode />);

    expect(asFragment()).toMatchSnapshot();
  });
});
