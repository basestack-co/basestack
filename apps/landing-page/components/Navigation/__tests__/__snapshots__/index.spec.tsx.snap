// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation tests should render Navigation in dark mode 1`] = `
<DocumentFragment>
  .c5 {
  height: 2rem;
  width: 2rem;
  flex-shrink: 0;
}

.c6 {
  color: #000000;
  font-size: 2rem;
}

.c9 {
  height: 2.625rem;
  padding: 0 1rem;
  font-size: 1rem;
  position: relative;
  border: none;
  display: flex;
  align-items: center;
  font-weight: 500;
  cursor: pointer;
  border-radius: 4px;
  text-decoration: none;
  transition: background-color 0.1s ease-in-out,color 0.1s ease-in-out,border-color 0.1s ease-in-out;
  background-color: #000000;
  color: #FFFFFF;
  width: medium;
  height: medium;
  background-color: transparent;
  width: fit-content;
}

.c9:hover:not(:active, :disabled) {
  background-color: #333;
}

.c9 .material-symbols-sharp {
  color: inherit;
}

.c9:disabled {
  cursor: not-allowed;
  opacity: 0.6;
}

.c11 {
  height: 2.625rem;
  padding: 0 1rem;
  font-size: 1rem;
  position: relative;
  border: none;
  display: flex;
  align-items: center;
  font-weight: 500;
  cursor: pointer;
  border-radius: 4px;
  text-decoration: none;
  transition: background-color 0.1s ease-in-out,color 0.1s ease-in-out,border-color 0.1s ease-in-out;
  background-color: #000000;
  color: #FFFFFF;
  width: medium;
  height: medium;
  margin-right: 0.75rem;
  background-color: transparent;
  width: fit-content;
}

.c11:hover:not(:active, :disabled) {
  background-color: #333;
}

.c11 .material-symbols-sharp {
  color: inherit;
}

.c11:disabled {
  cursor: not-allowed;
  opacity: 0.6;
}

.c12 {
  height: 2.625rem;
  padding: 0 1rem;
  font-size: 1rem;
  position: relative;
  border: none;
  display: flex;
  align-items: center;
  font-weight: 500;
  cursor: pointer;
  border-radius: 4px;
  text-decoration: none;
  transition: background-color 0.1s ease-in-out,color 0.1s ease-in-out,border-color 0.1s ease-in-out;
  background-color: #000000;
  color: #FFFFFF;
  width: medium;
  height: medium;
  width: fit-content;
}

.c12:hover:not(:active, :disabled) {
  background-color: #333;
}

.c12 .material-symbols-sharp {
  color: inherit;
}

.c12:disabled {
  cursor: not-allowed;
  opacity: 0.6;
}

.c4 {
  margin-right: 0.75rem;
  border: none;
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
  border-radius: 50%;
  transition: background-color 0.1s ease-in-out,color 0.1s ease-in-out,border-color 0.1s ease-in-out;
  background-color: transparent;
  color: #000000;
  height: 2.5rem;
  width: 2.5rem;
}

.c4 .material-symbols-sharp {
  color: inherit;
}

.c4:hover {
  background-color: #EEEEEE;
}

.c4:active {
  background-color: #E2E2E2;
}

.c0 {
  position: sticky;
  top: 0;
  z-index: 1000;
  display: flex;
  flex-shrink: 0;
  align-items: center;
  height: 5rem;
  padding: 0 1.25rem;
  transition: all 0.2s ease-in;
  background: rgba(31, 31, 31, 0.6);
  -webkit-backdrop-filter: saturate(180%) blur(20px);
  backdrop-filter: saturate(180%) blur(20px);
}

.c1 {
  display: flex;
  align-items: center;
  margin: 0 auto;
  width: 100%;
  max-width: 1440px;
}

.c2 {
  display: flex;
  align-items: center;
}

.c7 {
  display: flex;
  margin-left: 0.75rem;
}

.c8 {
  margin-right: 0.25rem;
}

.c10 {
  display: flex;
  margin-left: auto;
}

.c3>button {
  background-color: transparent;
  color: #CBCBCB;
  transition: background-color 0.1s ease-in-out;
}

.c3>button:hover {
  background-color: #333333;
}

@media screen and (max-width: 991.98px) {
  .c7 {
    display: none;
  }
}

@media screen and (min-width: 992px) {

}

@media screen and (min-width: 992px) {
  .c3 {
    display: none;
  }
}

<nav
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <button
            class="c4"
          >
            <div
              class="c5"
              data-testid="icon-container"
              size="32"
            >
              <span
                class="c6 material-symbols-sharp"
                color="#000000"
                data-testid="icon"
                size="32"
              >
                menu
              </span>
            </div>
          </button>
        </div>
        <svg
          fill="none"
          height="42"
          viewBox="0 0 36 36"
          width="42"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="black"
            height="36"
            rx="4"
            width="36"
          />
          <g
            clip-path="url(#clip0_3_10)"
          >
            <path
              d="M20.4 12L20.16 10.8C20.07 10.34 19.66 10 19.18 10H12C11.45 10 11 10.45 11 11V26C11 26.55 11.45 27 12 27C12.55 27 13 26.55 13 26V20H18.6L18.84 21.2C18.93 21.67 19.34 22 19.82 22H25C25.55 22 26 21.55 26 21V13C26 12.45 25.55 12 25 12H20.4Z"
              fill="white"
            />
          </g>
          <defs>
            <clippath
              id="clip0_3_10"
            >
              <rect
                fill="white"
                height="24"
                transform="translate(6 6)"
                width="24"
              />
            </clippath>
          </defs>
        </svg>
        <ul
          class="c7"
        >
          <li
            class="c8"
          >
            <button
              class="c9"
              data-testid="button"
            >
              <span
                class=""
              >
                Platform
              </span>
            </button>
          </li>
          <li
            class="c8"
          >
            <button
              class="c9"
              data-testid="button"
            >
              <span
                class=""
              >
                Discover Features
              </span>
            </button>
          </li>
          <li
            class="c8"
          >
            <button
              class="c9"
              data-testid="button"
            >
              <span
                class=""
              >
                Integration
              </span>
            </button>
          </li>
          <li
            class="c8"
          >
            <button
              class="c9"
              data-testid="button"
            >
              <span
                class=""
              >
                Why Feature Flags?
              </span>
            </button>
          </li>
          <li
            class="c8"
          >
            <button
              class="c9"
              data-testid="button"
            >
              <span
                class=""
              >
                FAQs
              </span>
            </button>
          </li>
        </ul>
      </div>
      <div
        class="c10"
      >
        <button
          class="c11"
          data-testid="button"
        >
          <span
            class=""
          >
            Github
          </span>
        </button>
        <button
          class="c12"
          data-testid="button"
        >
          <span
            class=""
          >
            Documentation
          </span>
        </button>
      </div>
    </div>
  </nav>
  .c0 {
  position: fixed;
  z-index: 999;
  top: 79px;
  left: 0;
  right: 0;
  bottom: 0;
  overflow: hidden;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  pointer-events: none;
}

.c1 {
  display: flex;
  flex-direction: column;
  background: rgba(31, 31, 31, 0.6);
  -webkit-backdrop-filter: saturate(180%) blur(20px);
  backdrop-filter: saturate(180%) blur(20px);
  padding: 0.75rem;
}

@media screen and (max-width: 991.98px) {

}

@media screen and (min-width: 992px) {
  .c0 {
    display: none;
  }
}

@media screen and (min-width: 992px) {

}

<div
    class="c0"
  >
    <ul
      class="c1"
      style="opacity: 0; width: 0px; height: 0px;"
    />
  </div>
</DocumentFragment>
`;

exports[`Navigation tests should renders Navigation correctly 1`] = `
<DocumentFragment>
  .c5 {
  height: 2rem;
  width: 2rem;
  flex-shrink: 0;
}

.c6 {
  color: #000000;
  font-size: 2rem;
}

.c9 {
  height: 2.625rem;
  padding: 0 1rem;
  font-size: 1rem;
  position: relative;
  border: none;
  display: flex;
  align-items: center;
  font-weight: 500;
  cursor: pointer;
  border-radius: 4px;
  text-decoration: none;
  transition: background-color 0.1s ease-in-out,color 0.1s ease-in-out,border-color 0.1s ease-in-out;
  background-color: #EEEEEE;
  color: #000000;
  width: medium;
  height: medium;
  background-color: transparent;
  width: fit-content;
}

.c9:hover:not(:active, :disabled) {
  background-color: #E2E2E2;
}

.c9 .material-symbols-sharp {
  color: inherit;
}

.c9:disabled {
  cursor: not-allowed;
  opacity: 0.6;
}

.c11 {
  height: 2.625rem;
  padding: 0 1rem;
  font-size: 1rem;
  position: relative;
  border: none;
  display: flex;
  align-items: center;
  font-weight: 500;
  cursor: pointer;
  border-radius: 4px;
  text-decoration: none;
  transition: background-color 0.1s ease-in-out,color 0.1s ease-in-out,border-color 0.1s ease-in-out;
  background-color: #EEEEEE;
  color: #000000;
  width: medium;
  height: medium;
  margin-right: 0.75rem;
  background-color: transparent;
  width: fit-content;
}

.c11:hover:not(:active, :disabled) {
  background-color: #E2E2E2;
}

.c11 .material-symbols-sharp {
  color: inherit;
}

.c11:disabled {
  cursor: not-allowed;
  opacity: 0.6;
}

.c12 {
  height: 2.625rem;
  padding: 0 1rem;
  font-size: 1rem;
  position: relative;
  border: none;
  display: flex;
  align-items: center;
  font-weight: 500;
  cursor: pointer;
  border-radius: 4px;
  text-decoration: none;
  transition: background-color 0.1s ease-in-out,color 0.1s ease-in-out,border-color 0.1s ease-in-out;
  background-color: #000000;
  color: #FFFFFF;
  width: medium;
  height: medium;
  width: fit-content;
}

.c12:hover:not(:active, :disabled) {
  background-color: #333;
}

.c12 .material-symbols-sharp {
  color: inherit;
}

.c12:disabled {
  cursor: not-allowed;
  opacity: 0.6;
}

.c4 {
  margin-right: 0.75rem;
  border: none;
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
  border-radius: 50%;
  transition: background-color 0.1s ease-in-out,color 0.1s ease-in-out,border-color 0.1s ease-in-out;
  background-color: transparent;
  color: #000000;
  height: 2.5rem;
  width: 2.5rem;
}

.c4 .material-symbols-sharp {
  color: inherit;
}

.c4:hover {
  background-color: #EEEEEE;
}

.c4:active {
  background-color: #E2E2E2;
}

.c0 {
  position: sticky;
  top: 0;
  z-index: 1000;
  display: flex;
  flex-shrink: 0;
  align-items: center;
  height: 5rem;
  padding: 0 1.25rem;
  transition: all 0.2s ease-in;
  background: rgba(238, 238, 238, 0.7);
  -webkit-backdrop-filter: saturate(180%) blur(20px);
  backdrop-filter: saturate(180%) blur(20px);
}

.c1 {
  display: flex;
  align-items: center;
  margin: 0 auto;
  width: 100%;
  max-width: 1440px;
}

.c2 {
  display: flex;
  align-items: center;
}

.c7 {
  display: flex;
  margin-left: 0.75rem;
}

.c8 {
  margin-right: 0.25rem;
}

.c10 {
  display: flex;
  margin-left: auto;
}

@media screen and (max-width: 991.98px) {
  .c7 {
    display: none;
  }
}

@media screen and (min-width: 992px) {

}

@media screen and (min-width: 992px) {
  .c3 {
    display: none;
  }
}

<nav
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <button
            class="c4"
          >
            <div
              class="c5"
              data-testid="icon-container"
              size="32"
            >
              <span
                class="c6 material-symbols-sharp"
                color="#000000"
                data-testid="icon"
                size="32"
              >
                menu
              </span>
            </div>
          </button>
        </div>
        <svg
          fill="none"
          height="42"
          viewBox="0 0 36 36"
          width="42"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="black"
            height="36"
            rx="4"
            width="36"
          />
          <g
            clip-path="url(#clip0_3_10)"
          >
            <path
              d="M20.4 12L20.16 10.8C20.07 10.34 19.66 10 19.18 10H12C11.45 10 11 10.45 11 11V26C11 26.55 11.45 27 12 27C12.55 27 13 26.55 13 26V20H18.6L18.84 21.2C18.93 21.67 19.34 22 19.82 22H25C25.55 22 26 21.55 26 21V13C26 12.45 25.55 12 25 12H20.4Z"
              fill="white"
            />
          </g>
          <defs>
            <clippath
              id="clip0_3_10"
            >
              <rect
                fill="white"
                height="24"
                transform="translate(6 6)"
                width="24"
              />
            </clippath>
          </defs>
        </svg>
        <ul
          class="c7"
        >
          <li
            class="c8"
          >
            <button
              class="c9"
              data-testid="button"
            >
              <span
                class=""
              >
                Platform
              </span>
            </button>
          </li>
          <li
            class="c8"
          >
            <button
              class="c9"
              data-testid="button"
            >
              <span
                class=""
              >
                Discover Features
              </span>
            </button>
          </li>
          <li
            class="c8"
          >
            <button
              class="c9"
              data-testid="button"
            >
              <span
                class=""
              >
                Integration
              </span>
            </button>
          </li>
          <li
            class="c8"
          >
            <button
              class="c9"
              data-testid="button"
            >
              <span
                class=""
              >
                Why Feature Flags?
              </span>
            </button>
          </li>
          <li
            class="c8"
          >
            <button
              class="c9"
              data-testid="button"
            >
              <span
                class=""
              >
                FAQs
              </span>
            </button>
          </li>
        </ul>
      </div>
      <div
        class="c10"
      >
        <button
          class="c11"
          data-testid="button"
        >
          <span
            class=""
          >
            Github
          </span>
        </button>
        <button
          class="c12"
          data-testid="button"
        >
          <span
            class=""
          >
            Documentation
          </span>
        </button>
      </div>
    </div>
  </nav>
  .c0 {
  position: fixed;
  z-index: 999;
  top: 79px;
  left: 0;
  right: 0;
  bottom: 0;
  overflow: hidden;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  pointer-events: none;
}

.c1 {
  display: flex;
  flex-direction: column;
  background: rgba(238, 238, 238, 0.7);
  -webkit-backdrop-filter: saturate(180%) blur(20px);
  backdrop-filter: saturate(180%) blur(20px);
  padding: 0.75rem;
}

@media screen and (max-width: 991.98px) {

}

@media screen and (min-width: 992px) {
  .c0 {
    display: none;
  }
}

@media screen and (min-width: 992px) {

}

<div
    class="c0"
  >
    <ul
      class="c1"
      style="opacity: 0; width: 0px; height: 0px;"
    />
  </div>
</DocumentFragment>
`;
